### ISSUE TITLE
matching in catch clauses

### ISSUE THREAD

**Issue by opax** [eXist-db/exist#764]
The matching of QNames in catch clauses seems to ignore the actually bound namespaces (eXist 2.2):

In this query, the error is caught although it is in a different namespace:

``` xquery
xquery version "3.0";
declare namespace ns = "urn:ns";
declare namespace ns2 = "urn:ns2";

try {
    error(QName("urn:ns2", "ns:error"))
} catch ns:error {
    'caught_in_wrong_namespace'
} catch * {
    'wildcard'
}
```

While in this query, the error is not caught although it should be:

``` xquery
xquery version "3.0";
declare namespace ns = "urn:ns";
declare namespace ns2 = "urn:ns2";

try {
    error(QName("urn:ns", "some_prefix:error"))
} catch ns:error {
    'caught'
} catch * {
    'should_have_been_caught'
}
```

This is probably related to issues #743, #744, #745

**Comment by opax** [comment]
This seemed to be fixed in 3.0 and can be closed. 

I did write XQSuite tests demonstrating the problem (which is how I found out that it is fixed). Should I put those in a pull request?

**Comment by joewiz** [comment]
@opax Great idea, yes please! Thanks in advance. And if this issue is ok to close, please go ahead and close it.
